### PR TITLE
Add main endpoint for job

### PR DESCRIPTION
**Pull request by erebe**

### CODE REVIEW

File: src/resources/job/EnvironmentJob.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+get:
+  summary: 'List jobs'
+  operationId: listJob
```
Comment (mzottola) [inline src/resources/job/EnvironmentJob.yaml:3]:
```suggestion
  operationId: listJobs
```

File: src/openapi.yaml [hunk 2]
```diff
@@ -632,6 +662,38 @@ paths:
...
+  #/job/{jobId}/secret/{secretId}/override:
+  #/job/{jobId}/environmentVariable/{environmentVariableId}/alias:
+  #/job/{jobId}/secret/{secretId}/alias:
+  #/job/{jobId}/environmentVariable/{environmentVariableId}:
+  #/job/{jobId}/environmentVariable/{variableId}:
+  #/job/{jobId}/secret:
+  #/job/{jobId}/secret/{secretId}:
+  #/job/{jobId}/environmentVariable/import:
+  #/organization/{organizationId}/job/preview:
+  #/organization/{organizationId}/job/deploy:
```
Comment (mzottola) [inline src/openapi.yaml:696]: We don't expose those endpoints yet ?

File: src/schemas/_index.yaml [hunk 3]
```diff
@@ -362,8 +362,20 @@ InvoiceResponseList:
...
+  $ref: ./job/JobResponseList.yaml
+JobAdvancedSettings:
+  $ref: ./job/JobAdvancedSettings.yaml
+JobDeployRequest:
+  $ref: ./job/JobDeployRequest.yaml
+DeploymentHistoryJobResponse:
+  $ref: ./job/DeploymentHistoryJobResponse.yaml
 Key:
-  $ref: ./Key.yaml
+ $ref: ./Key.yaml
```
Comment (mzottola) [inline src/schemas/_index.yaml:378]: Typo
```suggestion
  $ref: ./Key.yaml
```

File: src/schemas/job/JobResponse.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        type: boolean
+        description: |
+          Indicates if the 'environment preview option' is enabled for this container.  
+          If enabled, a preview environment will be automatically cloned when `/preview` endpoint is called.  
+          If not specified, it takes the value of the `auto_preview` property from the associated environment.
+      port:
+        type: integer
+        nullable: true
+        minimum: 1
+        description: Port were to run readiness and liveliness probes checks. The port will not be exposed externally
```
Comment (mzottola) [inline src/schemas/job/JobResponse.yaml:69]:
```suggestion
        description: Port where to run readiness and liveliness probes checks. The port will not be exposed externally
```

File: src/schemas/job/JobRequest.yaml [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        type: boolean
+        description: |
+          Indicates if the 'environment preview option' is enabled for this container.  
+          If enabled, a preview environment will be automatically cloned when `/preview` endpoint is called.  
+          If not specified, it takes the value of the `auto_preview` property from the associated environment.
+      port:
+        type: integer
+        nullable: true
+        minimum: 1
+        description: Port were to run readiness and liveliness probes checks. The port will not be exposed externally
```
Comment (mzottola) [inline src/schemas/job/JobRequest.yaml:47]:
```suggestion
        description: Port where to run readiness and liveliness probes checks. The port will not be exposed externally
```

File: src/schemas/job/JobResponse.yaml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      - maximum_cpu
+      - maximum_memory
+    properties:
+      environment:
+        $ref: '../ReferenceObject.yaml'
+      registry:
+        $ref: '../ReferenceObject.yaml'
+      maximum_cpu:
+        type: integer
+        description: Maximum cpu that can be allocated to the container based on organization cluster configuration. unit is millicores (m). 1000m = 1 cpu
```
Comment (mzottola) [inline src/schemas/job/JobResponse.yaml:24]:
```suggestion
        description: Maximum cpu that can be allocated to the job based on organization cluster configuration. unit is millicores (m). 1000m = 1 cpu
```

File: src/schemas/job/JobResponse.yaml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        example: 16000
+      maximum_memory:
+        type: integer
+        description: Maximum memory that can be allocated to the container based on organization cluster configuration. unit is MB. 1024 MB = 1GB
```
Comment (mzottola) [inline src/schemas/job/JobResponse.yaml:28]:
```suggestion
        description: Maximum memory that can be allocated to the job based on organization cluster configuration. unit is MB. 1024 MB = 1GB
```

File: src/schemas/_index.yaml [hunk 8]
```diff
@@ -362,8 +362,20 @@ InvoiceResponseList:
   $ref: ./InvoiceResponseList.yaml
```
Comment (mzottola) [inline src/schemas/_index.yaml:362]: Some structure is missing, e.g JobScheduleEvent

File: src/openapi.yaml (hunk 2 above)
Comment (erebe) [reply src/openapi.yaml:696]: Because I need to make the schemas/resources :see_no_evil: and they are not yet there

**Pull request merged**